### PR TITLE
Add gatk4 preprocessintervals module and respective test files

### PR DESCRIPTION
**Pull request by ryanjameskennedy**
## PR checklist

Gatk4 PreprocessIntervals' module was added along with the respective test files. It was also tested using `nf-core modules create-test-yml nf-core/gatk4/preprocessintervals`.

- [x] This comment contains a description of changes (with reason).
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/gatk4/preprocessintervals/main.nf [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        log.info '[GATK PreprocessIntervals] Available memory not known - defaulting to 3GB. Specify process memory requirements to change this.'
+    } else {
+        avail_mem = task.memory.giga
+    }
+    """
+    gatk --java-options "-Xmx${avail_mem}g" PreprocessIntervals \\
+        $exclude_command \\
+        --reference $fasta \\
+        --padding 0 \\
+        -imr OVERLAPPING_ONLY \\
```
Comment (ramprasadn) [inline modules/nf-core/gatk4/preprocessintervals/main.nf:39]: These parameters should be sent in via ext.args from a config file, so I'd suggest removing them from here.

File: modules/nf-core/gatk4/preprocessintervals/main.nf [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        --output preprocessed_intervals.interval_list \\
```
Comment (ramprasadn) [inline modules/nf-core/gatk4/preprocessintervals/main.nf:40]: I'd suggest building the output file name from prefix variable like we do in other modules [ex](https://github.com/nf-core/modules/blob/00646879a0d08b2fdad809be29c4848de9886e75/modules/nf-core/abacas/main.nf#L29)

File: modules/nf-core/gatk4/preprocessintervals/meta.yml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        Groovy Map containing sample information
+        e.g. [ id:'test', single_end:false ]
+  - versions:
+      type: file
+      description: File containing software versions
+      pattern: "versions.yml"
+  - interval_list:
+      type: file
+      description: Processed interval list file
+      pattern: "*.{bed,interval_list}"
```
Comment (ramprasadn) [inline modules/nf-core/gatk4/preprocessintervals/meta.yml:55]: remove bed

File: modules/nf-core/gatk4/preprocessintervals/meta.yml [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+input:
+  - meta:
+      type: map
+      description: |
+        Groovy Map containing sample information
+        e.g. [ id:'test', single_end:false ]
+  - exclude_intervals:
+      type: file
+      description: Bed file with the genomic regions to be excluded from the analysis (optional)
+      pattern: "*.{bed,interval_list}"
```
Comment (ramprasadn) [inline modules/nf-core/gatk4/preprocessintervals/meta.yml:28]: Can you provide a bed file to this option? Or does it only accept interval_list? 
And if the tool doesn't accept a bed file, then modify the description too.

File: modules/nf-core/gatk4/preprocessintervals/meta.yml (hunk 3 above)
Comment (ramprasadn) [reply modules/nf-core/gatk4/preprocessintervals/meta.yml:55]: don't forget this one as well.

File: modules/nf-core/gatk4/preprocessintervals/meta.yml (hunk 4 above)
Comment (ryanjameskennedy) [reply modules/nf-core/gatk4/preprocessintervals/meta.yml:28]: It can take both interval_list files and bed files. I've tested both.

Comment (ramprasadn) [reply modules/nf-core/gatk4/preprocessintervals/meta.yml:28]: 👍🏻

**Review by ramprasadn — APPROVED** [review]
LGTM 👍🏻

**Pull request merged**